packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+import { Provider } from '../zkEvm/types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+  passportProvider: Provider | null;
+  web3Provider: Web3Provider | null;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<string[]>;
+  loginCallback: () => Promise<void>;
+};
+
+const PassportContext = createContext<PassportContextType | null>(null);
+
+type PassportProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function PassportProvider({ children, config }: PassportProviderProps) {
```

Comment:
It would be useful if we could pass in the passport instance, rather than the config.

Reply:
Reason: when using checkout, you want to be able to pass the passport instance to checkout. Obviously you can still get it by calling `usePassport` but it makes it more tricky as you would want to initialise checkout and passport together